### DISCUSSION TITLE
ImageGraphics

### DISCUSSION THREAD

**Discussion by tusharsable** [iTwin/itwinjs-core#2384] [Q&A] [answered]
Q1. We have some analytical data as an image . We want to show this image in the model at places we want. We found out that it can be done with ImageGraphic class. Please share a sample code for doing so.

 
Q2. How to upload an image texture to iModelDb using (Texture.createTexture() method) how can we create IModelDb object as a parameter for the createTexture function. Please share a sample code for doing so.

 
Q3. Can we change the transparency of the image graphics we create by the above method. Please share a sample code for doing so.

**Comment by pmconne** [comment]
It sounds like you want to decorate your viewport with these images, rather than adding them as persistent elements to your iModel.
If that's the case, you'll need to write a [Decorator](https://www.itwinjs.org/learning/frontend/viewdecorations/).
Assuming you have your image as an HTMLImageElement, you can produce a RenderTexture from it like this:
```
const texture = IModelApp.renderSystem.createTextureFromImage(image, false, iModelConnection, new RenderTexture.Params());
```
You can then create a material that will apply that texture to some geometry:
```
const materialParams = new RenderMaterial.Params();
materialParams.textureMapping = new TextureMapping(texture, new TextureMapping.Params());
const material = IModelApp.renderSystem.createMaterial(materialParams, iModelConnection);
```
Your decorator's `decorate` method will receive a `DecorateContext` to which you can add some textured geometry:
```
public decorate(ctx: DecorateContext): void {
  const builder = ctx.createGraphicBuilder(GraphicType.WorldDecoration); // or WorldOverlay if you want it to be visible in front of other geometry.
  const params = new GraphicParams();
  params.material = material; // The one you created above
  builder.activeGraphicParams(params); // Any geometry subsequently added to the builder will use your textured material
  // Assuming you want a rectangular image and `points` describes the rectangle:
  builder.addShape(points);
  // Add the decoration to the scene
  ctx.addDecorationFromBuilder(builder);
}
```

Here's [one example](https://www.itwinjs.org/sandboxes/ThomasDicarlo/World%20Decoration%20Markers) that draws "markers" using images mapped to ellipses. See SmartDeviceMarkerDecoration.tsx.
Here's [another example](https://www.itwinjs.org/sample-showcase/?group=Viewer+Features&sample=heatmap-decorator-sample&imodel=Metrostation+Sample) that draws an image produced from a heatmap. See HeatmapDecorator.ts.

If further assistance is required please provide more details.

**Comment by tomdicarlo** [comment, marked as answer]
To expand on what @pmconne wrote, I put together a [sandbox](https://www.itwinjs.org/sandboxes/ThomasDicarlo/Image%20Mapping%20to%20Geometry) that implements a solution like the one he describes, along with a walkthrough highlighting and explaining the code segments relevant to your question. 

Feel free to use any of the code provided, and let me know if you have any feedback about how I could improve the sandbox to better answer your question.
